tests/test_observation.py

```diff
@@ -234,3 +234,54 @@ def test_index_extraction(self):
             observation.images[bands],
             Image(np.array([image_g, image_r, image_i]), bands=("g", "r", "i")),
         )
+
+    def test_slicing(self):
+        np.random.seed(25)
+        all_bands = tuple("grizy")
+        images = np.random.random((5, 11, 13))
+        variance = np.arange(5)[:, None, None] * np.ones((5, 11, 13)) + 1
+        weights = 1 / variance
+        psfs = np.arange(5)[:, None, None] * np.ones((5, 2, 2))
+        model_psf = np.zeros(9).reshape(3, 3)
+        model_psf[1] = 1
+        model_psf[:, 1] = 1
+        model_psf[1, 1] = 2
+        bbox = Box((11, 13), origin=(12, 10))
+        observation = Observation(
+            images,
+            variance,
+            weights,
+            psfs,
+            model_psf[None],
+            bands=all_bands,
+            bbox=bbox,
+        )
+
+        new_box = Box((3, 4), origin=(15, 13))
+        sliced_bands = tuple("riz")
+        sliced_observation = observation[sliced_bands, new_box]
+        self.assertImageEqual(
+            sliced_observation.images["r":"z"],
+            Image(images[1:4, 3:6, 3:7], bands=sliced_bands, yx0=new_box.origin),
+        )
+        self.assertImageEqual(
+            sliced_observation.variance["r":"z"],
+            Image(variance[1:4, 3:6, 3:7], bands=sliced_bands, yx0=new_box.origin),
+        )
+        self.assertImageEqual(
+            sliced_observation.weights["r":"z"],
+            Image(weights[1:4, 3:6, 3:7], bands=sliced_bands, yx0=new_box.origin),
+        )
+        np.testing.assert_array_almost_equal(
```

Comment:
Why are these almost_equal?

Reply:
They actually are equal in this case. In general I use `almost_equal` anytime I have an array of floats because sometimes numpy can be weird. In this case I created the psfs as an integer array but internally `Observation.__init__` ensures that the images, weights, and psfs have the same dtype, and at least once in the past this caused `assert_array_equal` to fail. But if you prefer I can change this to `assert_array_equal` because it seems to work now with this version of numpy.

Reply:
Ok. numpy recommends using `assert_allclose` instead of `array_almost_equal` ([link](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_array_almost_equal.html)). If you do need to use it for roundoff errors I recommend passing a smaller-than-default rtol and atol as otherwise the tolerances are too generous.